types/row_proof_test.go

```diff
@@ -59,6 +59,26 @@ func TestRowProofValidate(t *testing.T) {
 			root:    root,
 			wantErr: true,
 		},
+		{
+			name: "invalid start row",
+			rp: func() RowProof {
+				proof := validRowProof()
+				proof.StartRow += 1
+				return proof
+			}(),
+			root:    root,
+			wantErr: true,
```

Comment:
nit: this test could be improved by verifying that the error is `fmt.Errorf("invalid start/end row")` in case some other validation check is erroring before the expected error is encountered.

Reply:
I think that's fine as long as we're attesting that some error happens. Changing this would mean refactoring the whole test + it's not something that's being done consistently in this repo, most tests are just wanting an error.

If you feel strongly about it, I can refactor, no issue

Reply:
Not blocking on it. It's just a nice to have